heaps/heap_sort.py

```diff
@@ -1,8 +1,22 @@
+from heaps.min_heap import MinHeap
 
 
 def heap_sort(list):
     """ This method uses a heap to sort an array.
-        Time Complexity:  ?
-        Space Complexity: ?
+        Time Complexity:  O(n * n log n)
+        Space Complexity: O(n)
     """
-    pass
\ No newline at end of file
+    if len(list) <= 1:
+        return list
+
+    heap = MinHeap()
+
+    for item in list:
+        heap.add(item)
+
+    return_list = []
```

Comment:
Your approach looks good and should work if you complete the `heap_down` method.